src/Services/FhirResourcesService.php

```diff
@@ -80,13 +81,12 @@ public function createPatientResource($resourceId = '', $data = '', $encode = tr
         $address->setState($data['state']);
         $address->setPostalCode($data['postal_code']);
         $gender->setValue(strtolower($data['sex']));
-
+        $initResource['name'] = array();
+        array_push($initResource['name'], $name->jsonSerialize());
+        $initResource['address'] = array();
+        array_push($initResource['address'], $address->jsonSerialize());
+        $initResource['sex'] = strtolower($data['sex']);
```

Comment:
I don't get what you're trying to do here. This will miss gender and active state.

Reply:
> I don't get what you're trying to do here. This will miss gender and active state.

Ohh I missed to add the gender and active state.

> Is it to tedious to look at the classes to determine methods for building resource? why the need to serialize certain elements?

I'm passing the $initResource to the constructor of the resources rather than calling each of the function of the resource individually. 
$address and $name are objects; the constructor accepts arrays only. The jsonSerialize() methods are already defined in the FhirAddress and FhirHumanName R4 classes which help in converting these objects to arrays.

Reply:
@sjpadgett Sir, if you don't want these changes, I can revert.

Reply:
No it's fine if this is the way you want to do it. I get what you did. I just don't get why. It was already working the way it was. 
Go ahead and fix gender and active and i'll look closer at the others in a.m. Please when testing take a look at any produced json for structure and content.

Reply:
> Go ahead and fix gender and active
Yeah, I'll fix that.

> Please when testing take a look at any produced json for structure and content
Yes, I always do a double check on that.

Reply:
I still prefer using R4 class methods to build resources. One reason is it reads better. Also, will help if we support a version change.

> $address and $name are objects; the constructor accepts arrays only. The jsonSerialize() methods are already defined in the FhirAddress and FhirHumanName R4 classes which help in converting these objects to arrays.

 I don't understand still why you want to handle this way when all that is need is addName() and addAddress().  Let the resource class do that work.